File: app/gui/design/tokens/typography.py
"""
SOC-IQ Design System
Typography Design Tokens
"""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class TextStyle:
    """Immutable typography style."""

    family: str
    size: int
    weight: int


class FontFamily:
    PRIMARY = "Segoe UI"
    MONOSPACE = "JetBrains Mono"


class FontWeight:
    LIGHT = 300
    REGULAR = 400
    MEDIUM = 500
    SEMIBOLD = 600
    BOLD = 700


class FontSize:
    DISPLAY = 28
    HEADING = 22
    TITLE = 18
    SUBTITLE = 16
    BODY = 11
    BODY_SMALL = 10
    LABEL = 10
    CAPTION = 9
    CODE = 10


class LineHeight:
    TIGHT = 1.1
    NORMAL = 1.3
    RELAXED = 1.5


class Typography:
    """Semantic typography tokens."""

    DISPLAY = TextStyle(
        FontFamily.PRIMARY,
        FontSize.DISPLAY,
        FontWeight.BOLD,
    )

    HEADING = TextStyle(
        FontFamily.PRIMARY,
        FontSize.HEADING,
        FontWeight.SEMIBOLD,
    )

    TITLE = TextStyle(
        FontFamily.PRIMARY,
        FontSize.TITLE,
        FontWeight.SEMIBOLD,
    )

    SUBTITLE = TextStyle(
        FontFamily.PRIMARY,
        FontSize.SUBTITLE,
        FontWeight.MEDIUM,
    )

    BODY = TextStyle(
        FontFamily.PRIMARY,
        FontSize.BODY,
        FontWeight.REGULAR,
    )

    BODY_SMALL = TextStyle(
        FontFamily.PRIMARY,
        FontSize.BODY_SMALL,
        FontWeight.REGULAR,
    )

    LABEL = TextStyle(
        FontFamily.PRIMARY,
        FontSize.LABEL,
        FontWeight.MEDIUM,
    )

    CAPTION = TextStyle(
        FontFamily.PRIMARY,
        FontSize.CAPTION,
        FontWeight.REGULAR,
    )

    CODE = TextStyle(
        FontFamily.MONOSPACE,
        FontSize.CODE,
        FontWeight.REGULAR,
    )

File: frontend/src/styles/tokens/typography.ts
/**
 * SOC-IQ Design System — Typography Tokens
 * Ported 1:1 from `app/gui/design/tokens/typography.py`.
 *
 * Note: `FontFamily.primary` ("Segoe UI") is a Windows system font. It
 * is kept as the ported value (not invented here) but should resolve
 * through a web-safe stack at CSS-variable definition time — see
 * `styles/globals.css`. Not a target-state exception, just a rendering
 * detail; the token value itself is unchanged from Python.
 */

export const FontFamily = {
  primary: "Segoe UI",
  monospace: "JetBrains Mono",
} as const;

export const FontWeight = {
  light: 300,
  regular: 400,
  medium: 500,
  semibold: 600,
  bold: 700,
} as const;

export const FontSize = {
  display: 28,
  heading: 22,
  title: 18,
  subtitle: 16,
  body: 12,
  bodySmall: 11,
  label: 11,
  caption: 10,
  code: 10,
} as const;

export const LineHeight = {
  tight: 1.1,
  normal: 1.3,
  relaxed: 1.5,
} as const;

export interface TextStyle {
  family: string;
  size: number;
  weight: number;
}

function textStyle(family: string, size: number, weight: number): TextStyle {
  return { family, size, weight };
}

export const Typography = {
  display: textStyle(FontFamily.primary, FontSize.display, FontWeight.bold),
  heading: textStyle(
    FontFamily.primary,
    FontSize.heading,
    FontWeight.semibold,
  ),
  title: textStyle(FontFamily.primary, FontSize.title, FontWeight.semibold),
  subtitle: textStyle(
    FontFamily.primary,
    FontSize.subtitle,
    FontWeight.medium,
  ),
  body: textStyle(FontFamily.primary, FontSize.body, FontWeight.regular),
  bodySmall: textStyle(
    FontFamily.primary,
    FontSize.bodySmall,
    FontWeight.regular,
  ),
  label: textStyle(FontFamily.primary, FontSize.label, FontWeight.medium),
  caption: textStyle(
    FontFamily.primary,
    FontSize.caption,
    FontWeight.regular,
  ),
  code: textStyle(FontFamily.monospace, FontSize.code, FontWeight.regular),
} as const;

export type TypographyToken = typeof Typography;
